### PR TITLE
feat(dropdown): v2

### PR DESCRIPTION
**Pull request by pagrawl3**
All new dropdown thats a lot more stable

### CODE REVIEW

File: src/components/Inputs/Dropdown/Dropdown.js [hunk 1]
```diff
@@ -1,151 +1,137 @@
...
-    updatedOptions,
-    searchKeys || [optionKey],
-  );
-  const useDropdownProps = [
-    searchTerm,
-    searchResults,
+  // stores the currently typed input (in case of withSearch)
+  const [text, setText] = React.useState("");
+  const [options, setOptions] = React.useState(propOptions);
+  const [createdOption, setCreatedOption] = React.useState(undefined);
```
Comment (kartik0709) [inline src/components/Inputs/Dropdown/Dropdown.js]: no need to define undefined :P

File: src/components/Inputs/Dropdown/Dropdown.js [hunk 2]
```diff
@@ -1,151 +1,137 @@
...
-  const [searchTerm, searchResults, handleSearch] = useSearch(
-    updatedOptions,
-    searchKeys || [optionKey],
-  );
-  const useDropdownProps = [
-    searchTerm,
-    searchResults,
+  // stores the currently typed input (in case of withSearch)
+  const [text, setText] = React.useState("");
+  const [options, setOptions] = React.useState(propOptions);
```
Comment (kartik0709) [inline src/components/Inputs/Dropdown/Dropdown.js]: isn't options already equal to propOptions ^

File: src/components/Inputs/Dropdown/components/Input/Input.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from "react";
+import PropTypes from "prop-types";
+
+import Icon from "@src/components/UI/Icon";
+import AngleDownIcon from "@src/icons/angle-down.svg";
```
Comment (kartik0709) [inline src/components/Inputs/Dropdown/components/Input/Input.js]: No need for import, you can just use string `"angle-down"`

File: src/components/Inputs/Dropdown/Dropdown.js (hunk 2 above)
Comment (pagrawl3) [reply src/components/Inputs/Dropdown/Dropdown.js]: no? options is a newly created state value.

File: src/components/Inputs/Dropdown/components/Input/Input.js (hunk 3 above)
Comment (pagrawl3) [reply src/components/Inputs/Dropdown/components/Input/Input.js]: this is generally a better practice, reduces dependency.

**Pull request merged**